docs/components/ButtonWithOptions.md

```diff
@@ -49,16 +56,24 @@ Object containing the secondary options. Keys define the text of the option and
  ```jsx
  <ButtonWithOptions
    onClick={this.mainCallback()}
-   options={{
-     'Save and continue': this.saveAndContinueCallback()
-     'Save and go back': this.saveAndGoBackeCallback()
-   }}
+   options={new Map(
```

Comment:
I understand that a Map is semantically more correct than just a plain object, but I'm not keen on the complexity it adds (lengthier declaration + `Array.from(map.keys())`). The reason we went with an object in the first place was for simplicity of directly mapping keys (labels) to values (callbacks). If you think an object is ugly in this case, I'd rather go with an array containing objects with `label` and `callback` properties.

What do you think? Was there any other reason for this change that I might be overlooking?

Reply:
Changed to use the array form of the map I was using - hope that's ok